deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,406 @@
+"""Module containing robustness report check."""
+from collections import defaultdict
+
+import imgaug
+from typing import TypeVar, List, Optional, Any, Sized
+import albumentations
+
+import pandas as pd
+import torch
+from plotly.subplots import make_subplots
+import plotly.graph_objects as go
+from ignite.metrics import Metric
+
+from deepchecks import CheckResult, ConditionResult
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision import VisionData, SingleDatasetCheck, Context
+from deepchecks.vision.dataset import TaskType
+from deepchecks.vision.metrics_utils import calculate_metrics, metric_results_to_df
+from deepchecks.vision.utils.validation import set_seeds
+from deepchecks.vision.metrics_utils import get_scorers_list
+from deepchecks.utils.strings import format_percent
+from deepchecks.vision.utils.base_formatters import BasePredictionFormatter
+from deepchecks.vision.utils.image_functions import numpy_to_image_figure, apply_heatmap_image_properties, \
+    label_bbox_add_to_figure, get_image_info
+
+
+__all__ = ['RobustnessReport']
+
+
+PR = TypeVar('PR', bound='RobustnessReport')
+
+
+class RobustnessReport(SingleDatasetCheck):
+    """Check several image enhancements for model robustness.
+
+    Parameters
+    ----------
+        alternative_metrics (List[Metric], default None):
+            A list of ignite.Metric objects whose score should be used. If None are given, use the default metrics.
+    """
+
+    def __init__(self,
+                 prediction_formatter: BasePredictionFormatter,
+                 alternative_metrics: Optional[List[Metric]] = None,
+                 augmentations=None,
+                 random_state: int = 42):
+        super().__init__()
+        self.alternative_metrics = alternative_metrics
+        self.prediction_formatter = prediction_formatter
+        self.random_state = random_state
+        self.augmentations = augmentations
```

Comment:
probably should be validated here?

Reply:
what should be? the type of augmentations? anyway there is validation needed that is based on the dataset that passed, so felt it is too much to have separate validations